otter/metrics.py

```diff
@@ -277,15 +279,8 @@ def collect_metrics(reactor, config, client=None, authenticator=None,
             total_desired, total_actual, total_pending))
 
     # Add to cloud metrics
-
-    # WARNING: This request func is configured to make requests against the
-    # metrics region. The same request_func can't be used for other purposes,
-    # like making requests to cloud servers.
-    req_func = get_request_func(authenticator, config['metrics']['tenant_id'],
-                                metrics_log, service_configs,
-                                config['metrics']['region'])
     eff = add_to_cloud_metrics(
```

Comment:
Thanks, that's a very important catch! Fixed and pushed.

The tests for it aren't very nice; collect_metrics is really pretty nasty and is definitely deserving a refactoring to be more Effectful, but the fact that get_all_metrics is still returning a Deferred and not an Effect (because of the whole parallel limit thing) is a blocker. I may try to factor this stuff better over the next couple weeks.